crm_compassion/models/crm_lead.py

```diff
@@ -81,3 +81,39 @@ def _compute_name(self):
         for lead in self:
             if not lead.name and lead.partner_id and lead.partner_id.name:
                 lead.name = lead.partner_id.name
+
+    @api.model
+    def _read_group_stage_ids(self, stages, domain, order):
+        team_id = self._context.get("default_team_id")
+
+        # default behavior of parent
+        if team_id:
+            search_domain = [
+                "|",
+                ("id", "in", stages.ids),
+                "|",
+                ("team_id", "=", False),
+                ("team_id", "=", team_id),
+            ]
+        else:
+            search_domain = ["|", ("id", "in", stages.ids), ("team_id", "=", False)]
+
+        # if the domain contains team_id filters, add them to the search domain
+        team_id_domain = [
+            cond
+            for cond in domain
+            if hasattr(cond, "__getitem__") and cond[0] == "team_id"
+        ]
+        if len(team_id_domain) > 0:
+            search_domain = [
+                "|",
+                *search_domain,
+                *(["|"] * (len(team_id_domain) - 1) + team_id_domain),
+            ]
+
+        stage_ids = stages._search(
+            search_domain, order=order, access_rights_uid=SUPERUSER_ID
+        )
+        return stages.browse(stage_ids) + super(CrmLead, self)._read_group_stage_ids(
+            stages, domain, order
+        )
```

Comment:
As the super() method already returns records, can't we simplify the inheritance here? I'm not sure it is equivalent, but something like this?
```suggestion
        res = super()._read_group_stage_ids(stages, domain, order)
        # if the domain contains team_id filters, add them to the search domain
        team_id_domain = [
            cond
            for cond in domain
            if hasattr(cond, "__getitem__") and cond[0] == "team_id"
        ]
        if len(team_id_domain) > 0:
            search_domain = [
                *(["|"] * (len(team_id_domain) - 1) + team_id_domain),
            ]
            res += self.browse(self._search(
                search_domain, order=order, access_rights_uid=SUPERUSER_ID
            ))
        return res
```